### PR TITLE
Add new command to resolve the commit associated behind the build id

### PR DESCRIPTION
**Pull request by xTVaser**
Main limitation is not being able to resolve very old build ids (ie. the ones from v1.5).  But that can be improved in the future.

image

Writing java outside of work feels...weird

### CODE REVIEW

File: src/main/java/io/github/redpanda4552/HifumiBot/HifumiBot.java [hunk 1]
```diff
@@ -138,6 +145,13 @@ public HifumiBot() {
         dynCmdConfig = (DynCmdConfig) ConfigManager.read(ConfigType.DYNCMD);
         ConfigManager.write(dynCmdConfig);
 
+        ConfigManager.createConfigIfNotExists(ConfigType.BUILDMAP);
+        buildCommitMap = (BuildCommitMap) ConfigManager.read(ConfigType.BUILDMAP);
+        if (buildCommitMap != null) {
+            buildCommitMap.seedMap();
+        }
+        ConfigManager.write(dynCmdConfig);
```
Comment (RedPanda4552) [inline src/main/java/io/github/redpanda4552/HifumiBot/HifumiBot.java:153]: Copy-paste error? Meant to write buildCommitMap?

File: src/main/java/io/github/redpanda4552/HifumiBot/command/slash/CommandBuildNumber.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    public CommandBuildNumber() {
+        super(PermissionLevel.GUEST);
+    }
+
+    @Override
+    protected void onExecute(SlashCommandEvent event) {
+        OptionMapping option = event.getOption("buildid");
+        if (option == null) {
+            event.reply("Build ID not provided or invalid!").queue();
```
Comment (RedPanda4552) [inline src/main/java/io/github/redpanda4552/HifumiBot/command/slash/CommandBuildNumber.java:60]: Could chain a .setEphemeral() after .reply(String) to limit this message to just the sender.

File: src/main/java/io/github/redpanda4552/HifumiBot/command/slash/CommandBuildNumber.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        Optional<String> commitSha = HifumiBot.getSelf().getBuildCommitMap().getCommitSha(buildId);
+        commitSha.ifPresentOrElse((sha) -> {
+            EmbedBuilder eb = new EmbedBuilder();
+            eb.setTitle("Build Id to Commit");
+            eb.addField("Build Id", String.valueOf(buildId), true);
+            eb.addField("Commit", String.format("[%s](https://github.com/PCSX2/pcsx2/commit/%s)",
+                    StringUtils.abbreviate(sha, 10), sha), true);
+            event.replyEmbeds(eb.build()).queue();
+        }, () -> {
+            event.reply("Could not find that build id in the current development cycle (v1.7)!").queue();
```
Comment (RedPanda4552) [inline src/main/java/io/github/redpanda4552/HifumiBot/command/slash/CommandBuildNumber.java:73]: Ditto but I'd be more accepting of leaving this one as a public reply.

File: src/main/resources/fixture-build-map.json [hunk 4]
```diff
@@ -0,0 +1,849 @@
+{
```
Comment (RedPanda4552) [inline src/main/resources/fixture-build-map.json:1]: Looks like this is a list of all the old system's build numbers + commit hashes yeah?

Comment (xTVaser) [reply src/main/resources/fixture-build-map.json:1]: yeah up to 1801 for now

File: src/main/java/io/github/redpanda4552/HifumiBot/command/slash/CommandBuildNumber.java (hunk 2 above)
Comment (xTVaser) [reply src/main/java/io/github/redpanda4552/HifumiBot/command/slash/CommandBuildNumber.java:60]: thats actually exactly what i wanted to do but couldnt find how to do it, ill make that change

**Pull request merged**